pkgs/development/python-modules/pytorch/default.nix

```diff
@@ -174,33 +201,44 @@ in buildPythonPackage rec {
     ninja
   ] ++ lib.optionals cudaSupport [ cudatoolkit_joined ];
 
-  buildInputs = [
-    numpy.blas
-  ] ++ lib.optionals cudaSupport [ cudnn magma nccl ]
+  buildInputs = [(if mklSupport then mkl else openblas)]
+    ++ lib.optionals cudaSupport [ cudnn magma nccl ]
     ++ lib.optionals stdenv.isLinux [ numactl ];
 
   propagatedBuildInputs = [
     cffi
     click
     numpy
     pyyaml
+    # the following are required for tensorboard support
+    pillow six future tensorflow-tensorboard protobuf
   ] ++ lib.optionals openMPISupport [ openmpi ]
-    ++ lib.optional (pythonOlder "3.5") typing
-    ++ lib.optionals tensorboardSupport [pillow six future tensorflow-tensorboard];
+    ++ lib.optional (pythonOlder "3.5") typing;
 
-  checkInputs = [ hypothesis ninja ];
+  checkInputs = [ hypothesis ninja psutil ];
 
   doCheck = false; # tests take a long time for channel release, so doCheck should be overridden only when developing
-  checkPhase = "${cudaStubEnv}python test/run_test.py"
-    + " --exclude utils" # utils requires git, which is not allowed in the check phase
-
-    # Other tests which have been disabled in previous nix derivations of pytorch.
-    # --exclude dataloader sparse torch utils thd_distributed distributed cpp_extensions
-    ;
+  checkPhase = with lib.versions; with lib.strings; concatStringsSep " " [
+    # MKL 2019.5-only workaround. See: https://github.com/NixOS/nixpkgs/issues/75611
+    (optionalString (mklSupport && majorMinor mkl.version == "2019.5") "KMP_INIT_AT_FORK=FALSE ")
```

Comment:
Master is now on `mkl-2020.1.217`; can you check if this bug is still around?  I hope we can delete this entirely.

Reply:
I have a few healthcare-related things going on right now and am not even sure I'll be able to go through too many more cycles of review -- this will need to be reviewed by the next person.